pkg/csi/service/fcd/vsphere_helper.go

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package fcd
+
+import (
+	"sort"
+	"strings"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	"golang.org/x/net/context"
+
+	cm "k8s.io/cloud-provider-vsphere/pkg/common/connectionmanager"
+	"k8s.io/cloud-provider-vsphere/pkg/common/vclib"
+)
+
+const (
+	NUM_OF_CONNECTION_ATTEMPTS     int = 3
+	RETRY_ATTEMPT_DELAY_IN_SECONDS int = 1
+)
+
+func removePortFromHost(host string) string {
+	result := host
+	index := strings.IndexAny(host, ":")
+	if index != -1 {
+		result = host[:index]
+	}
+	return result
+}
+
+// getAllFCDs returns all FCDs in all VC/DC sorted by UUID
+func getAllFCDs(ctx context.Context, cm *cm.ConnectionManager) []*vclib.FirstClassDiskInfo {
+
+	firstClassDisks := make([]*vclib.FirstClassDiskInfo, 0)
+
+	for vc, vsi := range cm.VsphereInstanceMap {
+
+		var err error
+		for i := 0; i < NUM_OF_CONNECTION_ATTEMPTS; i++ {
+			err = cm.ConnectByInstance(ctx, vsi)
+			if err == nil {
+				break
+			}
+			time.Sleep(time.Duration(RETRY_ATTEMPT_DELAY_IN_SECONDS) * time.Second)
```

Comment:
It's not. I was moving around code that was using it, but it isn't really needed anymore.

Reply:
I take that back... I tried removing, but it is complaining about it. Maybe because it's in a const? Oh well... will just leave it.

```
[go] invalid operation: RETRY_ATTEMPT_DELAY_IN_SECONDS * time.Second (mismatched types int and time.Duration)
```